core/src/ast/expr.rs

```diff
@@ -61,3 +61,303 @@ pub enum Expr {
+            &Expr::Extract {
+                field: DateTimeField::Minute,
+                expr: Box::new(Expr::Identifier("2022-05-05 01:02:03".to_string()))
+            }
+            .to_sql()
+        );
+
+        assert_eq!(
+            "id BETWEEN low AND high",
+            &Expr::Between {
+                expr: Box::new(Expr::Identifier("id".to_string())),
+                negated: false,
+                low: Box::new(Expr::Identifier("low".to_string())),
+                high: Box::new(Expr::Identifier("high".to_string()))
+            }
+            .to_sql()
+        );
+
+        assert_eq!(
+            "id NOT BETWEEN low AND high",
+            &Expr::Between {
+                expr: Box::new(Expr::Identifier("id".to_string())),
+                negated: true,
+                low: Box::new(Expr::Identifier("low".to_string())),
+                high: Box::new(Expr::Identifier("high".to_string()))
+            }
+            .to_sql()
+        );
+
+        assert_eq!(
+            r#"id IN ("a","b","c")"#,
+            &Expr::InList {
+                expr: Box::new(Expr::Identifier("id".to_string())),
+                list: vec![
+                    Expr::Literal(AstLiteral::QuotedString("a".to_string())),
+                    Expr::Literal(AstLiteral::QuotedString("b".to_string())),
+                    Expr::Literal(AstLiteral::QuotedString("c".to_string()))
+                ],
+                negated: false
+            }
+            .to_sql()
+        );
+
+        assert_eq!(
+            r#"id NOT IN ("a","b","c")"#,
+            &Expr::InList {
+                expr: Box::new(Expr::Identifier("id".to_string())),
+                list: vec![
+                    Expr::Literal(AstLiteral::QuotedString("a".to_string())),
+                    Expr::Literal(AstLiteral::QuotedString("b".to_string())),
+                    Expr::Literal(AstLiteral::QuotedString("c".to_string()))
+                ],
+                negated: true
+            }
+            .to_sql()
+        );
+
+        assert_eq!(
+            "CASE id\nWHEN 1 THEN \"a\"\nWHEN 2 THEN \"b\"\nELSE \"c\"\nEND",
+            &Expr::Case {
+                operand: Some(Box::new(Expr::Identifier("id".to_string()))),
+                when_then: vec![
+                    (
+                        Expr::Literal(AstLiteral::Number(BigDecimal::from_str("1").unwrap())),
+                        Expr::Literal(AstLiteral::QuotedString("a".to_string()))
+                    ),
+                    (
+                        Expr::Literal(AstLiteral::Number(BigDecimal::from_str("2").unwrap())),
+                        Expr::Literal(AstLiteral::QuotedString("b".to_string()))
+                    )
+                ],
+                else_result: Some(Box::new(Expr::Literal(AstLiteral::QuotedString(
+                    "c".to_string()
+                ))))
+            }
+            .to_sql()
+        );
```

Comment:
Test code readability is also important which we need to make it.
We could consider using regex to make it easier to read

e.g. you can change these to apply to other tests.
```rust
use regex::Regex;                                                                     
                                                                                      
let re = Regex::new(r"\n\s+").unwrap();                                               
let trim = |s: &str| re.replace_all(s.trim(), "\n").into_owned();                     
                                                                                      
assert_eq!(                                                                           
    trim(                                                                             
        r#"                                                                           
        CASE id                                                                       
          WHEN 1 THEN "a"                                                             
          WHEN 2 THEN "b"                                                             
          ELSE "c"                                                                    
        END                                                                           
        "#                                                                            
    ),                                                                                
    Expr::Case {                                                                      
        operand: Some(Box::new(Expr::Identifier("id".to_string()))),                  
        when_then: vec![                                                              
            (                                                                         
                Expr::Literal(AstLiteral::Number(BigDecimal::from_str("1").unwrap())),
                Expr::Literal(AstLiteral::QuotedString("a".to_string()))              
            ),                                                                        
            (                                                                         
                Expr::Literal(AstLiteral::Number(BigDecimal::from_str("2").unwrap())),
                Expr::Literal(AstLiteral::QuotedString("b".to_string()))              
            )                                                                         
        ],                                                                            
        else_result: Some(Box::new(Expr::Literal(AstLiteral::QuotedString(            
            "c".to_string()                                                           
        ))))                                                                          
    }                                                                                 
    .to_sql()                                                                         
);                                                                                    
```

Reply:
good idea.